src/test/java/wooteco/subway/domain/fare/FareCalculatorTest.java

```diff
@@ -0,0 +1,63 @@
+package wooteco.subway.domain.fare;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import wooteco.subway.domain.Line;
+import wooteco.subway.domain.Lines;
+import wooteco.subway.domain.fare.ageStrategy.AgeDiscountPolicy;
+import wooteco.subway.domain.fare.ageStrategy.ChildrenDiscountPolicy;
+import wooteco.subway.domain.fare.ageStrategy.NormalAgeDiscountPolicy;
+import wooteco.subway.domain.fare.ageStrategy.PreferentialDiscountPolicy;
+import wooteco.subway.domain.fare.ageStrategy.TeenagerDiscountPolicy;
+import wooteco.subway.domain.fare.distanceStrategy.DistanceDiscountPolicy;
+import wooteco.subway.domain.fare.distanceStrategy.ExtraDiscountPolicy;
+import wooteco.subway.domain.fare.distanceStrategy.FreeDiscountPolicy;
+import wooteco.subway.domain.fare.distanceStrategy.NormalDistanceDiscountPolicy;
+
+class FareCalculatorTest {
+
+    @ParameterizedTest(name = "{index} : {0}")
+    @MethodSource("parameters")
+    @DisplayName("거리, 노선들의 추가요금, 나이에 따라 요금을 계산한다.")
+    void calculateFare(
+        String testName,
+        AgeDiscountPolicy ageDiscountPolicy,
+        DistanceDiscountPolicy distanceDiscountPolicy,
+        int distance,
+        int expected) {
+
+        Lines lines = new Lines(List.of(
+            new Line("1호선", "red", 100),
+            new Line("2호선", "green", 500),
+            new Line("3호선", "orange", 300)
+        ));
+        FareCalculator fareCalculator = new FareCalculator(ageDiscountPolicy, distanceDiscountPolicy);
+        double actual = fareCalculator.calculate(distance, lines);
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    private static Stream<Arguments> parameters() {
+        return Stream.of(
+            Arguments.of("우대 + 거리 무료 + 노선추가요금500 일때 5km : 0원", new PreferentialDiscountPolicy(), new FreeDiscountPolicy(), 5, 0),
```

Comment:
해당 테스트에서는 Named.of를 사용하지 않은 특별한 이유가 있을까요?

Reply:
`Named.of()`의 파라미터가 `name`, `object` 여서 사용하지 못했습니다.

객체로 Object를 감싸고 getter로 꺼내오는 방법을 생각해봤는데 오히려 testName 파라미터를 줄이기 위해 더 가독성이 안좋아진다고 생각했습니다.

Reply:
```suggestion
            Arguments.of(Named.of("우대 + 거리 무료 + 노선추가요금500 일때 5km : 0원", new PreferentialDiscountPolicy()), new FreeDiscountPolicy(), 5, 0),
```

위와 같이 데이터를 하나만 넣고 나머지는 그대로 작성해도 동작하지만, Named.of를 사용하면 개발할 때 괄호를 조금 고민하여 작성해야할 것 같네요 ㅎ_ㅎ..